schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/spi/BalanceEndpoint.java

```diff
@@ -69,7 +69,7 @@
 public class BalanceEndpoint {
 
     @ApiMethod(name = "balance.createPayment", path = "balance/create-payment", httpMethod = ApiMethod.HttpMethod.POST)
-    public JasPaymentResponse createPayment(User caller, JasPaymentRequest paymentRequest) throws UnauthorizedException, PaymentException, BadRequestException {
+    public JasPaymentResponse createPayment(User caller, JasPaymentRequest paymentRequest) throws UnauthorizedException, EntityNotFoundException, PaymentException, BadRequestException {
```

Comment:
In the Endpoint methods, any exception that is not one of the exceptions under `com.google.api.server.spi.response` (UnauthorizedException, ForbiddenException, **NotFoundException**, etc) gets converted to `InternalServerErrorException` (500).  This is bad since the client calling the method doesn't know what happened.

So rather than throwing `EntityNotFoundException` here, it would be better to catch it, and throw a `NotFoundException` (404) which tells the client exactly what happened.